compiler-core/src/analyse.rs

```diff
@@ -342,33 +343,47 @@ pub fn register_import(
         }
     }
 
-    if unqualified.is_empty() {
-        // When the module has no unqualified imports, we track its usage
-        // so we can warn if not used by the end of the type checking
-        let _ = environment
-            .unused_modules
-            .insert(module_name.clone(), *location);
-    }
+    match alias {
+        Some((AssignName::Variable(alias), span)) if !unqualified.is_empty() => {
```

Comment:
We've checking whether the module is aliased multiple times here, but then using a `module_name` value which exists regardless of whether it is discarded. I think this module name should be optional and then we don't have the same check multiple times and potential for invalid data when one exists but the other does not.